ui/src/main/java/org/eclipse/starter/ui/Project.java

```diff
@@ -1,104 +1,191 @@
-        artifactId, profile.getProfile(), version);
-
-    var context = FacesContext.getCurrentInstance();
-    var response = (HttpServletResponse) context.getExternalContext().getResponse();
-    context.responseComplete();
-    try {
-      final var redirectUrl = getApplicationUri()
-          .map(root -> root + downloadUrl)
-          .orElse(downloadUrl);
-      response.sendRedirect(redirectUrl + "?faces-redirect=true");
-    } catch (IOException e) {
-      logger.log(SEVERE, "Was not able to generate file", e);
-      throw new RuntimeException(e);
-    }
-  }
-
-  private Optional<String> getApplicationUri() {
-    try {
-      var ext = FacesContext.getCurrentInstance().getExternalContext();
-      var uri = new URI(ext.getRequestScheme(),
-          null, ext.getRequestServerName(), ext.getRequestServerPort(),
-          ext.getRequestContextPath(), null, null);
-      return Optional.of(uri.toASCIIString());
-    } catch (URISyntaxException e) {
-      logger.log(SEVERE, "Was not able to get context path", e);
-      return Optional.empty();
-    }
-  }
-}
+	private static final Logger LOGGER = Logger.getLogger(MethodHandles.lookup().lookupClass().getName());
+	private static final Map<String, String> RUNTIMES = Map.ofEntries(entry("GlassFish", "glassfish"),
+			entry("Open Liberty", "open-liberty"), entry("Payara", "payara"), entry("TomEE", "tomee"),
+			entry("WildFly", "wildfly"));
+
+	private double jakartaVersion = 10;
+	private String profile = "core";
+	private int javaVersion = 17;
+	private boolean docker = false;
+	private String runtime = "none";
+
+	public double getJakartaVersion() {
+		return jakartaVersion;
+	}
+
+	public void setJakartaVersion(double jakartaVersion) {
+		this.jakartaVersion = jakartaVersion;
+	}
+
+	public String getProfile() {
+		return profile;
+	}
+
+	public void setProfile(String profile) {
+		this.profile = profile;
+	}
+
+	public int getJavaVersion() {
+		return javaVersion;
+	}
+
+	public void setJavaVersion(int javaVersion) {
+		this.javaVersion = javaVersion;
+	}
+
+	public boolean isDocker() {
+		return docker;
+	}
+
+	public void setDocker(boolean docker) {
+		this.docker = docker;
+	}
+
+	public String getRuntime() {
+		return runtime;
+	}
+
+	public void setRuntime(String runtime) {
+		this.runtime = runtime;
+	}
+
+	public Map<String, String> getRuntimes() {
+		List<String> runtimes = new ArrayList<>(RUNTIMES.keySet());
+		Collections.shuffle(runtimes);
+
+		Map<String, String> shuffledRuntimes = new LinkedHashMap<>();
+		runtimes.forEach(r -> shuffledRuntimes.put(r, RUNTIMES.get(r)));
+
+		return shuffledRuntimes;
+	}
+
+	public void generate() {
+		LOGGER.log(Level.INFO,
+				"Generating project - Jakarta EE version: {0}, Jakarta EE profile: {1}, Java SE version: {2}, Docker: {3}, runtime: {4}",
+				new Object[] { jakartaVersion, profile, javaVersion, docker, runtime });
+
+		try {
+			FacesContext facesContext = FacesContext.getCurrentInstance();
```

Comment:
FacesContext, ExternalContext can be injected directly.

Reply:
Addressed.